core/src/utils/mod.rs

```diff
@@ -130,3 +130,13 @@ fn debug_assert_is_checked() {
 // ================================================================================================
 
 pub use miden_formatting::hex::{to_hex, DisplayHex, ToHex};
+
+// EITHER
+// ================================================================================================
+
+/// Generic container for a choice between two types.
+#[derive(Debug, Clone, Copy)]
```

Comment:
It doesn't look like this is used, so I would suggest removing it for now. I would pull in the `either` crate in the future if we determine having an `Either` type would be useful, as it provides a lot of conveniences you'd expect on a monadic type like this.

Reply:
Removed.

Note: I defined it instead of pulling in the `either` crate, since @bobbinth typically pushes for us to limit the amount of dependencies we bring in. We can decide on whether we implement it ourselves or pull in the `either` crate next time we need it.